chimera/R3BAsyChimera.cxx

```diff
@@ -0,0 +1,369 @@
+    Double_t fA_in = M_in / U_MEV;
+    Double_t fZ_in = gMC->TrackCharge();
+
+    fELoss += dE / 1000.; // back to GeV
+    
+    Double_t lightYield = gMC->Edep(); 
+    Double_t slow = gMC->Edep(); 
+
+    if (gMC->TrackCharge()!=0) {
+     Double_t birkC1Mod = 0;
+     // Apply correction for higher charge states
+     if (fBirkC0==1){
+      if (TMath::Abs(gMC->TrackCharge())>=2)
+        birkC1Mod=fBirkC1*1.;
+      else
+        birkC1Mod=fBirkC1;
+     }
+     if (gMC->TrackCharge()<0)birkC1Mod=0;
+
+     Double_t birkCS1Mod = 0;
+     // Apply correction for higher charge states
+     if (fBirkCS0==1){
+      if (TMath::Abs(gMC->TrackCharge())>=2)
+        birkCS1Mod=fBirkCS1*1.;
+      else
+        birkCS1Mod=fBirkCS1;
+     }
+     if (gMC->TrackCharge()<0)birkCS1Mod=0;//2
+
+     Double_t dedxcm=0.;
+     Double_t lightYieldxcm=0.;
+     if (gMC->TrackStep()>0)
+     {
+      dedxcm=1000.*gMC->Edep()/(gMC->TrackStep()*4.51);
+      lightYield=lightYield/(1.+birkC1Mod*dedxcm+fBirkC2*dedxcm*dedxcm);
+//1      Double_t slow =  lightYield ;
+      fLightYield=fLightYield+lightYield;
+//1      lightYieldxcm=lightYield/(gMC->TrackStep());
+//1      slow=slow/(1.+birkCS1Mod*lightYieldxcm+fBirkCS2*lightYieldxcm*lightYieldxcm);
+      slow=slow/(1.+birkCS1Mod*dedxcm+fBirkCS2*dedxcm*dedxcm);//2
+      fSlow=fSlow+slow;
+     }
+    }
+
+    if (dE > 0)
+    {
+
+        fNSteps++;
+
+        // Set additional parameters at exit of active volume. Create R3BSofToFWPoint.
+        if (gMC->IsTrackExiting() || gMC->IsTrackStop() || gMC->IsTrackDisappeared())
+        {
+
+            fTrackID = gMC->GetStack()->GetCurrentTrackNumber();
+            fParentTrackID = gMC->GetStack()->GetCurrentParentTrackNumber();
+            fVolumeID = vol->getMCid();
+            fDetCopyID = vol->getCopyNo();
+            fTrackPID = gMC->TrackPid();
+            fUniqueID = gMC->GetStack()->GetCurrentTrack()->GetUniqueID();
+
+            gMC->TrackPosition(fPosOut);
+            gMC->TrackMomentum(fMomOut);
+
+            if (fELoss == 0.)
+                return kFALSE;
+
+//            cout << fVolumeID << " * " << fDetCopyID << endl;
+//            getchar();
+            
+            AddPoint(fTrackID,
+                     fVolumeID,
+                     fDetCopyID,
+                     fZ_in,
+                     fA_in,
+                     TVector3(fPosIn.X(), fPosIn.Y(), fPosIn.Z()),
+                     TVector3(fPosOut.X(), fPosOut.Y(), fPosOut.Z()),
+                     TVector3(fMomIn.Px(), fMomIn.Py(), fMomIn.Pz()),
+                     TVector3(fMomOut.Px(), fMomOut.Py(), fMomOut.Pz()),
+                     fTime,
+                     fLength,
+                     fELoss,
+		             fLightYield,
+		             fSlow);
+
+            // Increment number of AsyChimeraPoints for this track
+            R3BStack* stack = (R3BStack*)gMC->GetStack();
+            stack->AddPoint(kCHIMERA);
+
+            // Cal Data for each plastic
```

Comment:
Please, remove the commented code!

Reply:
Please, update this class according to:
https://github.com/R3BRootGroup/sofia/blob/dev/tofwall/sim/R3BSofTofW.cxx